File: web/api/types/Task.ts
import type { File } from './File.ts'

export type Task = {
  /**
   * @type string
   */
  asignee: string | null
  /**
   * @type string, date-time
   */
  completed_at: string | null
  /**
   * @type string, date-time
   */
  created_at: string
  /**
   * @type string
   */
  description: string | null
  /**
   * @type string, date-time
   */
  due_date: string | null
  /**
   * @type array
   */
  files: File[] | null
  /**
   * @type string
   */
  id: string
  /**
   * @type string
   */
  status: string
  /**
   * @type string
   */
  title: string
  /**
   * @type string, date-time
   */
  updated_at: string
}

File: internal/postgres/ent/schema/task.go
package schema

import (
	"time"

	"entgo.io/ent"
	"entgo.io/ent/schema/edge"
	"entgo.io/ent/schema/field"
)

// Task holds the schema definition for the Task entity.
type Task struct {
	ent.Schema
}

func (Task) Fields() []ent.Field {
	return []ent.Field{
		field.String("id"),
		field.String("title"),
		field.String("description").
			Optional().
			Nillable(),
		field.Time("due_date").
			Optional().
			Nillable(),
		field.String("assignee_name").
			Optional().
			Nillable(),
		field.Time("created_at").
			Default(time.Now),
		field.Time("updated_at").
			Default(time.Now).
			UpdateDefault(time.Now),
		field.Time("completed_at").
			Optional().
			Nillable(),
		field.Enum("status").
			Values(
				"pending",
				"in_progress",
				"in_review",
				"completed",
			).
			Default("pending"),
	}
}

// Edges of the Task.
func (Task) Edges() []ent.Edge {
	return []ent.Edge{
		edge.To("files", File.Type),
	}
}
